### PR TITLE
Fix issue persisting entity mapped with non-nullable association 

### LINKED ISSUE #1077: Persist of entity with non nullable association causes UnsupportedOperationException
Mapping example:
```
@JoinColumn(name = "device_id", nullable = false)
@ManyToOne(fetch = FetchType.LAZY, targetEntity = Device.class, optional = false)
private Device device;
```

Setting either `nullable = false` or `optional = false` will cause the following exception during a persist:
```
Caused by: java.lang.UnsupportedOperationException: reactive persistence context
	at org.hibernate.reactive.engine.impl.ReactivePersistenceContextAdapter.getDatabaseSnapshot(ReactivePersistenceContextAdapter.java:76)
	at org.hibernate.engine.internal.ForeignKeys.isTransient(ForeignKeys.java:306)
	at org.hibernate.engine.internal.ForeignKeys$Nullifier.isNullifiable(ForeignKeys.java:230)
	at org.hibernate.engine.internal.ForeignKeys$Nullifier.access$000(ForeignKeys.java:36)
	at org.hibernate.engine.internal.ForeignKeys.collectNonNullableTransientEntities(ForeignKeys.java:414)
	at org.hibernate.engine.internal.ForeignKeys.findNonNullableTransientEntities(ForeignKeys.java:385)
	at org.hibernate.action.internal.AbstractEntityInsertAction.findNonNullableTransientEntities(AbstractEntityInsertAction.java:91)
	at org.hibernate.reactive.engine.ReactiveActionQueue.addInsertAction(ReactiveActionQueue.java:346)
	at org.hibernate.reactive.engine.ReactiveActionQueue.addAction(ReactiveActionQueue.java:331)
```

### PR DESCRIPTION
**Pull request by blafond**
Fixes #1077

Requires minor change to [NonNullableTransientEntities](https://github.com/hibernate/hibernate-orm/blob/798f9b05184bd0b14bdf13841b127ddef912a62d/hibernate-core/src/main/java/org/hibernate/engine/internal/NonNullableTransientDependencies.java#L27).

Merge after [PR](https://github.com/hibernate/hibernate-orm/pull/4772) is merged and after newer ORM build

### CODE REVIEW

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/engine/ReactiveActionQueue.java [hunk 1]
```diff
@@ -326,12 +325,13 @@ public void clear() {
 		}
 	}
 
-	public CompletionStage<Void> addAction(ReactiveEntityRegularInsertAction action) {
+	public CompletionStage<Void> addAction(ReactiveEntityInsertAction action) {
```
Comment (DavideD) [inline hibernate-reactive-core/src/main/java/org/hibernate/reactive/engine/ReactiveActionQueue.java:328]: Why did you change the signature of this method?

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+	@Entity(name = "Painting")
+	@Table(name = "painting")
+	public static class Painting {
+		@Id
+		public Long id;
+		String name;
+
+		@JoinColumn(name = "author_id", nullable = false)
+		@ManyToOne(fetch = FetchType.LAZY, targetEntity = Artist.class) //, optional = false)
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java:72]: Maybe you could add another association that has `optional = false`. This way we are sure that both settings are working

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		test( context, deleteEntities( "Painting", "Artist" ) );
+	}
+
+	@Test
+	public void testNonNullableSuccess(TestContext context) {
+		test( context,
+				openMutinySession().chain( session -> session
+								.createQuery( "from Artist", Artist.class )
+								.getSingleResult() )
+						// We are checking `.getPaintings()` to verify success in persistence
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java:59]: You are not checking anything, though. Not even that it's not null

**Review by DavideD — CHANGES_REQUESTED** [review]

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/engine/ReactiveActionQueue.java (hunk 1 above)
Comment (blafond) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/engine/ReactiveActionQueue.java:328]: mistake. fixed

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java (hunk 2 above)
Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java]: fixed

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java (hunk 3 above)
Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java]: fixed

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+							context.assertEquals( 1, paintings.size() );
+							context.assertEquals( "Mona Lisa", paintings.get( 0 ).name );
+						} )
+						.chain( () -> getMutinySessionFactory().withTransaction( s -> session
+										.createQuery( "from Dealer", Dealer.class )
+										.getSingleResult().chain( d -> session.fetch( d.paintings ) )
+										.invoke( paintings -> {
+											context.assertNotNull( paintings );
+											context.assertEquals( 1, paintings.size() );
+											context.assertEquals( "Mona Lisa", paintings.get( 0 ).name );
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java:74]: This indentation is weird. It makes it look like the second `withTransaction` is inside the first one (even if it's not the case).
It's not critical but since we have to wait for the next ORM release anyway, could you align them correctly, please? (Or maybe create two different methods)

**Review by DavideD — APPROVED** [review]

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java (hunk 4 above)
Comment (DavideD) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java:74]: Actually, something is wrong in the stream.

Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java:74]: fixed alone with correcting sessions in test.
